org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/generator/ResourceDescriptionStrategyFragmentTests.xtend

```diff
@@ -0,0 +1,136 @@
+	def testGenerate() {
+		val fragment = initializeFragmentWithGrammarFromString(TestableResourceDescriptionStrategyFragment, '''
+			grammar org.xtext.Foo with org.eclipse.xtext.common.Terminals
+			generate foo "http://org.xtext/foo"
+			Model: rules+=Rule;
+			@Exported
+			Rule: name=ID;
+			@Exported
+			Foo returns Rule: name=ID;
+		''')
+
+		val exportedRules = fragment.getExportedRulesFromGrammar
+		assertFalse(exportedRules.empty)
+		assertEquals(1, exportedRules.size)
+		assertTrue(fragment.shouldGenerateArtefacts(exportedRules))
+		val result = fragment.generateResourceDescriptionStrategyContent(fragment.superTypeRef, exportedRules)
+		assertEquals('''
+			public class FooAbstractResourceDescriptionStrategy extends org.eclipse.xtext.resource.impl.DefaultResourceDescriptionStrategy {
+				public boolean createEObjectDescriptions(interface org.eclipse.emf.ecore.EObject eObject, interface org.eclipse.xtext.util.IAcceptor<interface org.eclipse.xtext.resource.IEObjectDescription> acceptor) {
+					if(eObject instanceof org.xtext.foo.Rule) {
+						return createEObjectDescriptionsForRule(eObject, acceptor);
+					}
+					return true;
+				}
+			
+				protected boolean createEObjectDescriptionsForRule(interface org.eclipse.emf.ecore.EObject eObject, interface org.eclipse.xtext.util.IAcceptor<interface org.eclipse.xtext.resource.IEObjectDescription> acceptor) {
+					return super.createEObjectDescriptions(eObject, acceptor);
+				}
+			}
+		'''.toString, result.concatenationClientToString)
+	}
+
+	@Test
+	def testGenerate_NoStubs() {
+		val fragment = initializeFragmentWithGrammarFromString(TestableResourceDescriptionStrategyFragment, '''
+			grammar org.xtext.Foo with org.eclipse.xtext.common.Terminals
+			generate foo "http://org.xtext/foo"
+			Model: rules+=Rule;
+			@Exported
+			Rule: name=ID;
+			@Exported
+			Foo returns Rule: name=ID;
+		''')
+		fragment.generateStub = false
+		fragment.generateXtendStub = false
+		val exportedRules = fragment.getExportedRulesFromGrammar
+		assertFalse(exportedRules.empty)
+		assertEquals(1, exportedRules.size)
+		assertTrue(fragment.shouldGenerateArtefacts(exportedRules))
+		val result = fragment.generateResourceDescriptionStrategyContent(fragment.superTypeRef, exportedRules)
+		assertEquals('''
+			public class FooDefaultResourceDescriptionStrategy extends org.eclipse.xtext.resource.impl.DefaultResourceDescriptionStrategy {
+				public boolean createEObjectDescriptions(interface org.eclipse.emf.ecore.EObject eObject, interface org.eclipse.xtext.util.IAcceptor<interface org.eclipse.xtext.resource.IEObjectDescription> acceptor) {
+					if(eObject instanceof org.xtext.foo.Rule) {
+						return createEObjectDescriptionsForRule(eObject, acceptor);
+					}
+					return true;
+				}
+			
+				protected boolean createEObjectDescriptionsForRule(interface org.eclipse.emf.ecore.EObject eObject, interface org.eclipse.xtext.util.IAcceptor<interface org.eclipse.xtext.resource.IEObjectDescription> acceptor) {
```

Comment:
Expectation is bogus (interface org.eclipse... is not a valid java token at that location). Will be fixed when the JavaTypeAware ... is used.